test/bh/mods-test.js

```diff
@@ -0,0 +1,102 @@
+var fixtures = require('../fixtures')('bemhtml');
+var compile = fixtures.compile;
+
+describe('.mods()', function() {
+  var tmpls;
+  beforeEach(function() {
+    tmpls = compile(function() {
+      block('button').def()(function() {
+        return JSON.stringify(this.mods);
+      });
+      block('button').elem('*').def()(function() {
+        return JSON.stringify(this.mods);
+      });
+    });
+  });
+
+  it('should return empty mods', function() {
+    tmpls.apply({ block: 'button' }).should.equal('{}');
+  });
+
+  it('should return mods', function() {
+    tmpls.apply({ block: 'button', mods: { type: 'button' } })
+      .should.equal('{"type":"button"}');
+  });
+
+  xit('should return elem mods', function() {
+    tmpls.apply({ block: 'button', elem: 'control',
+        elemMods: { type: 'button' } })
+      .should.equal('{"type":"button"}');
+  });
+
+  it('should return boolean mods', function() {
+    tmpls.apply({ block: 'button', mods: { disabled: true } })
+      .should.equal('{"disabled":true}');
+  });
+
+  xit('should set mods', function() {
```

Comment:
Copy of [should support changing elemMods in runtime](https://github.com/bem/bem-xjst/blob/master/test/bemhtml/runtime-test.js#L418-L429)?